packages/calypso-e2e/src/lib/blocks/block-flows/shared/block-helpers.ts

```diff
@@ -0,0 +1,53 @@
+import { Locator, Page } from 'playwright';
+
+/**
+ * Makes a selector for a block based on the block name.
+ * Blocks often follow a predictable pattern using aria-labels, and this centralizes that!
+ *
+ * @param {string} blockName The name of the block, often how it appears in the sidebar.
+ * @returns A selector string to use to make a locator for the block.
+ */
+export function makeSelectorFromBlockName( blockName: string ): string {
+	return `div[aria-label="Block: ${ blockName }"]`;
+}
+
+export interface ExpectedFormField {
+	type: 'textbox' | 'checkbox' | 'radio' | 'combobox' | 'button';
+	accessibleName: string;
+}
+
+/**
+ * A shared validate function for published form blocks.
+ *
+ * @param {Locator | Page} publishedPage A locator/page object for the published page.
+ * @param {ExpectedFormField[]} expectedFormFields An array of fields to validate.
+ */
+export async function validatePublishedFormFields(
+	publishedPage: Locator | Page,
+	expectedFormFields: ExpectedFormField[]
+) {
+	for ( const expectedField of expectedFormFields ) {
+		const { type, accessibleName } = expectedField;
+		await publishedPage.getByRole( type, { name: accessibleName } ).first().waitFor();
+	}
+}
+
+interface FieldLabelDetails {
+	blockName: string;
+	accessibleLabelName: string;
+	labelText: string;
+}
```

Comment:
**suggestion:non-blocking** - can we consider moving the interface and type definitions up top, grouped together?